features/323-login-id/stage-1.md

```diff
@@ -0,0 +1,103 @@
+# Implict login ID key, Login ID realm
+
+## Database Schema
+
+- In `_auth_provider_password` table:
+    - A column `realm` (`TEXT NOT NULL`) would be added.
+    - The existing unique index on `login_id` would be changed to include `realm`.
+
+## Configuration
+
+- Add field `realmWhitelist` (list of strings, optional, default to `["default"]`).
+
+## HTTP API
+- `POST /login`:
+    - Change type of `login_id` from map of strings to a plain string value.
+    - Add parameter `login_id_key` (string, optional).
+    - Add parameter `realm` (string, optional).
+    - Example:
+      ```json
+        {
+            "login_id_key": "username",
+            "login_id": "test",
+            "password": "12345678"
+        }
+        {
+            "realm": "student",
+            "login_id": "test@example.com",
+            "password": "12345678"
+        }
+      ```
+- `POST /signup`
+    - Add parameter `realm` (string, optional).
+    - Example:
+      ```json
+        {
+            "realm": "teacher",
+            "login_ids": {
+                "email": "test@example.com"
+            },
+            "password": "12345678"
+        }
+        {
+            "realm": "student",
+            "login_id": {
+                "email": "test@example.com"
+            },
+            "password": "12345678"
+        }
+      ```
+
+- `POST /signup`, `POST /login`, etc.
+    - Remove list of login IDs from response. Method to obtain this list would
+      be specified in principal APIs later.
+
+- `POST /verify_request`
+    - Rename parameter `record_key` to `login_id_key`
+    - Add parameter `login_id` (string, optional).
+    - Example: 
+      ```json
+        {
+            "login_id_key": "email",
+            "login_id":"test@example.com"
+        }
+      ```
+
+
+## SDK API
```

Comment:
Try to avoid repeating the content to create maintainability troubles, my 2 cents.